packages/dds/tree/src/feature-libraries/sequence-field/rebase.ts

```diff
@@ -48,57 +48,60 @@ export type NodeChangeRebaser<TNodeChange> = (
     const baseIter = new StackyIterator(baseMarkList);
     const currIter = new StackyIterator(currMarkList);
-    for (let baseMark of baseIter) {
-        let currMark: Mark<TNodeChange> | undefined = currIter.pop();
-        if (currMark === undefined) {
-            break;
-        }
+    const lineageRequests: LineageRequest<TNodeChange>[] = [];
+    let baseDetachOffset = 0;
+    while (!baseIter.done || !currIter.done) {
+        let currMark: Mark<TNodeChange> | undefined = currIter.peek();
+        let baseMark: Mark<TNodeChange> | undefined = baseIter.peek();
 
-        if (isAttach(currMark)) {
-            // We currently ignore the ways in which base marks could affect attaches.
-            // These are:
-            // 1. Slices with which the attach would commute.
-            // 2. Attaches that target the same gap.
-            // We ignore #1 because slices are not yet supported.
-            // We ignore #2 because we do not yet support specifying the tiebreak.
-            factory.pushContent(clone(currMark));
-            baseIter.push(baseMark);
-        } else if (isReattach(currMark)) {
-            // We currently ignore the ways in which base marks could affect re-attaches.
-            // These are:
-            // 1. A reattach that targets the same tombs.
-            // 2. Attaches that target the same gap.
-            // We ignore #1 because it could only occur if undo were supported.
-            // We ignore #2 because we do not yet support specifying the tiebreak.
-            factory.pushContent(clone(currMark));
-            baseIter.push(baseMark);
-        } else if (isReattach(baseMark)) {
-            // We currently ignore the ways in which curr marks overlap with this re-attach.
-            // These are:
-            // 1. A reattach that matches this re-attach.
-            // 2. A tomb that matches this re-attach.
-            // We ignore #1 because it could only occur if undo were supported.
-            // We ignore #2 because we do not yet produce tombs.
-            factory.pushOffset(getOutputLength(baseMark));
-            currIter.push(currMark);
+        if (baseMark === undefined) {
+            assert(
+                currMark !== undefined,
+                "Loop condition should prevent both iterators from being empty",
+            );
+            if (baseDetachOffset > 0 && isAttach(currMark)) {
+                currIter.pop();
+                handleCurrAttach(currMark, factory, lineageRequests, baseDetachOffset);
+            } else {
+                break;
+            }
+        } else if (currMark === undefined) {
+            assert(
+                baseMark !== undefined,
+                "Loop condition should prevent both iterators from being empty",
+            );
+            baseIter.pop();
+            if (isRelevantToLineage(baseMark)) {
+                baseDetachOffset += getInputLength(baseMark);
+            } else {
+                break;
```

Comment:
Don't we need to set `baseDetachOffset` to zero here?

Reply:
Are we depending on it not being set to zero at line 135?

Reply:
Ok, it looks like we're just saving some lines of code by relying on the updateLineage that's after the loop. Closing this.